src/pages/companion/wrapped/firstConnection.tsx

```diff
@@ -0,0 +1,172 @@
+"use client";
+
+import NavBarContainer from "@/components/companion/navigation/NavBarContainer";
+import { CompanionConnectionRow } from "@/components/companion/connections/connection-row";
+import { useEffect, useState } from "react";
+import { fetchBackend } from "@/lib/db";
+import { Connection } from "@/components/companion/connections/connections-list";
+import { COMPANION_EMAIL_KEY } from "@/constants/companion";
+import { useRouter } from "next/navigation";
+import { motion, useMotionValue, animate } from 'framer-motion';
+import Image from 'next/image';
+
+interface FirstConnectionProps {
+  isPartner: boolean;
+}
+
+const FirstConnection = ({ isPartner }: FirstConnectionProps) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [connections, setConnections] = useState<Connection[]>([]);
+  const [error, setError] = useState("");
+  const [companyProfilePicture, setCompanyProfilePicture] = useState("");
+  const [searchQuery, setSearchQuery] = useState("");
+  const [isTapped, setIsTapped] = useState(false);
+  const router = useRouter();
+  const opacity = useMotionValue(1);
+  const scale = useMotionValue(1);
+  const y = useMotionValue(0);
+
+  const handleTap = () => {
+    if (connections.length > 0) {
+      localStorage.setItem("connections", JSON.stringify(connections));
+    }
+    setIsTapped(true);
+    animate(opacity, 0, { duration: 0.5 });
+    animate(scale, 0.8, { duration: 0.5 });
+    animate(y, 20, { duration: 0.5 });
+    setTimeout(() => {
+      router.push("/companion/wrapped/taps");
+    }, 800);
+  };
+
+  useEffect(() => {
+    const fetchConnections = async () => {
+      try {
+        setIsLoading(true);
+        const profileId = localStorage.getItem(COMPANION_EMAIL_KEY);
+        if (!profileId) {
+          setError("Please log in to view your connections");
+          setIsLoading(false);
+          return;
+        }
+
+        // Fetch connections using the profileID
+        const data = await fetchBackend({
+          endpoint: `/interactions/journal/${profileId}`,
+          method: "GET",
+          authenticatedCall: false,
+        });
+        setConnections(data.data);
+      } catch (error) {
+        console.error("Error fetching connections:", error);
+        setError("Error fetching your connections");
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchConnections();
+  }, []);
+
+  const fetchCompanyProfilePictures = async (company: string | undefined) => {
```

Comment:
This is very similar to the other `fetchCompanyProfilePictures` which expects an array, could we rename it to be singular? `fetchCompanyProfilePicture`